onnxruntime/core/session/onnxruntime_c_api.cc

```diff
@@ -2082,6 +2102,16 @@ ORT_API_STATUS_IMPL(OrtApis::CreateSessionFromArrayWithPrepackedWeightsContainer
   API_IMPL_END
 }
 
+
+ORT_API_STATUS_IMPL(OrtApis::GetValueMemoryInfo, const OrtValue* value, const OrtMemoryInfo** memory_info) {
+  API_IMPL_BEGIN
+  const auto& tensor = value->Get<onnxruntime::Tensor>();
+  *memory_info = &tensor.Location();
+  return nullptr;
```

Comment:
```suggestion
  *memory_info = nullptr;
  const auto& tensor = value->Get<onnxruntime::Tensor>();
  *memory_info = &tensor.Location();
```

Robust practice for out pointers is to nullify them up-front, in case an exception/early return happens before the value is assigned. That way it's not garbage in case of error. e.g. maybe `value->Get` throws, and `memory_info` is left uninitialized.